### PR TITLE
Add GitHub Actions workflow for issue milestone management

### PR DESCRIPTION
**Pull request by samsmithnz**
Add a new GitHub Actions workflow to manage issue milestones based on labels and issue state.

* **New Workflow File**: Add `.github/workflows/issue-milestone.yml` to handle issue labeling and milestone management.
* **Trigger Events**: Listen for `issues` events with `labeled` and `closed` actions.
* **Add Milestone**: Add "High Priority" milestone to issues labeled with "P1".
* **Remove Milestone**: Remove "High Priority" milestone from closed issues with the "P1" label.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/satisfactory-factories/application/pull/85?shareId=d6a44d1c-7535-4d00-bd88-8aa5b3aeb011).

### CODE REVIEW

**Review by Maelstromeous — CHANGES_REQUESTED** [review]

**Comment by Maelstromeous** [comment]
Love the planner though! Really need a nice "top fixes" list, and looks like the planner does the ticket!

File: .github/workflows/issue-milestone.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
```
[the remark this answers is not in the document]
Comment (samsmithnz) [reply .github/workflows/issue-milestone.yml]: Brain not on yet. Copilot bad!

**Comment by samsmithnz** [comment]
> Love the planner though! Really need a nice "top fixes" list, and looks like the planner does the ticket!

@Maelstromeous  Have you noticed that all the things we are working on are not 'high priority' though? We might want to think about either changing the priorities, or changing what we work on...

**Comment by Maelstromeous** [comment]
Yeah I'll fully admit I'm fully scatterbraining it at the moment 😅 But that is primarily down to the fact up until now I don't have a comprehensive list and I'm taking on the more interesting work, I do need to focus my efforts more I must admit.

Also it would be nice to possibly discuss prioritisation, I realise it is completely arbitary by me at the moment.

Thinking rather than using a P1,P2,P3 system we should use "Impact" and also a categorisation by Bug and Request. E.g. High Impact Bugs go right to the top, High impact features below that, medium impact bugs above medium impact features etc, what do yo think?

**Review by Maelstromeous — APPROVED** [review]
LGTM! 🚀

**Pull request merged**